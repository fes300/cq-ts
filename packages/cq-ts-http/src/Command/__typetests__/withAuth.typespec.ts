import * as t from 'io-ts';
import { Endpoint } from '..';
import { withAuth } from '../withAuth';

const endpoint = Endpoint({
  Input: {
    Params: { id: t.string },
    Headers: { id: t.string },
  },
  Method: 'GET',
  getPath: ({ id }) => `users/${id}/crayons`,
  Output: t.type({ crayons: t.array(t.string) }),
  Opts: { stringifyBody: true },
});

const noHeaderEndpoint = Endpoint({
  Input: {
    Params: { id: t.string },
  },
  Method: 'GET',
  getPath: ({ id }) => `users/${id}/crayons`,
  Output: t.type({ crayons: t.array(t.string) }),
  Opts: { stringifyBody: true },
});

const authenticatedEndpoint = withAuth(endpoint);

// @dts-jest:pass:snap when endpoint is wrapped other headers are still present
authenticatedEndpoint.Input.Headers.props.id;
// @dts-jest:pass:snap when endpoint is wrapped authorization header is present
authenticatedEndpoint.Input.Headers.props.authorization;

const authenticatedNoHeaderEndpoint = withAuth(noHeaderEndpoint);

// @dts-jest:pass:snap when endpoint with no previous headers is wrapped, authorization header is present
authenticatedNoHeaderEndpoint.Input.Headers.props.authorization;

// @dts-jest:fail:snap when endpoint is not wrapped you must not pass the header
noHeaderEndpoint.Input.Headers?.authorization;
