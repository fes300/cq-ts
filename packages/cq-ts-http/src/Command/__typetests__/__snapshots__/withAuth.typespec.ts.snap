// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when endpoint is not wrapped you must not pass the header (type) should match snapshot 1`] = `"Property 'authorization' does not exist on type 'never'."`;

exports[`when endpoint is wrapped authorization header is present (type) should match snapshot 1`] = `"BrandC<StringC, BearerTokenBrand>"`;

exports[`when endpoint is wrapped other headers are still present (type) should match snapshot 1`] = `"StringC"`;

exports[`when endpoint with no previous headers is wrapped, authorization header is present (type) should match snapshot 1`] = `"BrandC<StringC, BearerTokenBrand>"`;
